src/__tests__/buildTransfers.test.ts

```diff
@@ -0,0 +1,188 @@
+      expect(listed.data).to.be.equal(
+        "0xa9059cbb0000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000174876e800"
+      );
+
+      expect(unlisted.value).to.be.equal("0");
+      expect(unlisted.to).to.be.equal(unlistedToken.address);
+      expect(unlisted.data).to.be.equal(
+        "0xa9059cbb0000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000174876e800"
+      );
+
+      expect(native.value).to.be.equal(
+        fractional_amount.multipliedBy(TEN.pow(18)).toFixed()
+      );
+      expect(native.to).to.be.equal(receiverAddress);
+      expect(native.data).to.be.equal("0x");
+    });
+  });
+
+  describe("Mixed", () => {
+    it("works with arbitrary amount strings on listed, unlisted and native transfers", () => {
+      let amount = new BigNumber("123456.000000789");
+      let payments = listedUnlistedAndNativePayments(amount, receiverAddress);
+      let [listed, unlisted, native] = buildTransfers(
+        dummySafeInfo,
+        payments,
+        tokenList
+      );
+      expect(listed.value).to.be.equal("0");
+      expect(listed.to).to.be.equal(listedToken.address);
+      expect(listed.data).to.be.equal(
+        "0xa9059cbb0000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000001a24902beecbd5109200"
+      );
+
+      expect(unlisted.value).to.be.equal("0");
+      expect(unlisted.to).to.be.equal(unlistedToken.address);
+      expect(unlisted.data).to.be.equal(
+        "0xa9059cbb0000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000001a24902beecbd5109200"
+      );
+
+      expect(native.value).to.be.equal(
+        amount.multipliedBy(TEN.pow(18)).toFixed()
+      );
+      expect(native.to).to.be.equal(receiverAddress);
+      expect(native.data).to.be.equal("0x");
+    });
+  });
+
+  describe("Truncation on too many decimals", () => {
+    it("works with arbitrary amount strings on listed, unlisted and native transfers", () => {
+      let amount = new BigNumber("1.000000789");
+      const crappyToken: TokenInfo = {
+        address: "0x6b175474e89094c44da98b954eedeac495271d0f",
+        decimals: 0,
+        symbol: "NOD",
+        name: "No Decimals",
+        chainId: -1,
+      };
+      let payment: Payment = {
+        receiver: receiverAddress,
+        amount: amount,
+        tokenAddress: crappyToken.address,
+        decimals: crappyToken.decimals,
+      };
+      let [transfer] = buildTransfers(dummySafeInfo, [payment], tokenList);
+      expect(transfer.value).to.be.equal("0");
+      expect(transfer.to).to.be.equal(crappyToken.address);
+      expect(transfer.data).to.be.equal(
+        "0xa9059cbb00000000000000000000000010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001"
+      );
+    });
+  });
```

Comment:
I like the readability of this test alot. 

Just one important thing is missing.
So far we don't test error cases here and I think we also don't handle those cases well yet.

- too large amounts
- unlisted tokens without decimals set
- negative amounts although the csv parsing should cover this
- ...

Reply:
For unlisted tokens without decimals set... I would think this error should be caught during CSV parsing, but will include a test demonstrating what happens here anyway.

Reply:
I agree. I wanna build some more sophisticated CSV validation tonight.
Including a test for it.